### PR TITLE
Fixed some bugs in dlpy.metrics

### PR DESCRIPTION
**Pull request by sasxianhu**

### CODE REVIEW

File: dlpy/metrics.py [hunk 1]
```diff
@@ -154,18 +83,50 @@ def confusion_matrix(y_true, y_pred, castable=None, labels=None, id_vars=None):
                              row=y_true, col=y_pred)
     
     conf_mat = res['Crosstab']
-    
+
+    # make conf_mat to be a symmetric
+
+    # collect class info from rows and cols
+    row_class = [x.strip() for x in conf_mat.iloc[:,0].values]
```
Comment (ethem-kinginthenorth) [inline dlpy/metrics.py]: did we use any of these anywhere?

Comment (ethem-kinginthenorth) [reply dlpy/metrics.py]: actually I wrote this for line 95/96

File: dlpy/model.py [hunk 2]
```diff
@@ -268,14 +268,19 @@ def fit(self, data, inputs=None, target=None, data_specs=None, mini_batch_size=1
...
         else:
             print('NOTE: Training from scratch.')
             init_weights = None
 
+        # when model_weights is none, reset it
+        if self.model_weights is None:
+            self.model_weights = self.conn.CASTable('{}_weights'.format(self.model_name))
+
         if save_best_weights and self.best_weights is None:
             self.best_weights = random_name('model_best_weights', 6)
```
Comment (ethem-kinginthenorth) [inline dlpy/model.py]: it is quite difficult to see your changes, especially when you change the location of the functions :) github thinks you removed everything and added a bunch of things later.

File: dlpy/metrics.py (hunk 1 above)
Comment (sasxianhu) [reply dlpy/metrics.py]: no.

**Comment by ethem-kinginthenorth** [comment]
mergin

**Pull request merged**